### PR TITLE
docs: update getting started to account for async factory

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to update the `viem` package version to the latest compatible with `aa-sdk` and refactor the code related to creating a provider for sending user operations from a smart account.

### Detailed summary
- Updated `viem` package version to the latest compatible with `aa-sdk`
- Refactored code for creating a provider to send user operations from a smart account

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* **#498** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/498?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/498?utm_source=stack-comment).

File: site/snippets/getting-started/send-user-operation.ts [hunk 1]
```diff
@@ -6,22 +6,23 @@ import {
   type Hex,
 } from "@alchemy/aa-core";
 
+// select your chain from @alchemy/aa-core
```
Comment (avasisht23) [inline site/snippets/getting-started/send-user-operation.ts:9]: as opposed to viem

File: site/snippets/getting-started/send-user-operation.ts [hunk 2]
```diff
@@ -6,22 +6,23 @@ import {
...
-// Create a provider to send user operations from your smart account
-const provider = await createModularAccountAlchemyClient({
-  // get your Alchemy API key at https://dashboard.alchemy.com
-  apiKey: "ALCHEMY_API_KEY",
-  chain,
-  signer,
-});
-
 // [!code focus:22]
 (async () => {
```
Comment (denniswon) [inline site/snippets/getting-started/send-user-operation.ts:17]: what changed? or am i missing something

**Review by rthomare — APPROVED** [review]

File: site/snippets/getting-started/send-user-operation.ts (hunk 2 above)
Comment (denniswon) [reply site/snippets/getting-started/send-user-operation.ts:17]: I guess the original version assumes that the script allows top-level await. if not, await just not available.

Comment (avasisht23) [reply site/snippets/getting-started/send-user-operation.ts:17]: correct, the client factory is now async.

**Pull request merged**